src/charts/time_series_labeler.coffee

```diff
@@ -56,112 +85,156 @@ Ember.Charts.TimeSeriesLabeler = Ember.Mixin.create
       ticks
     else
       count = 1
-      interval = switch @get('selectedInterval')
-        when 'years', 'Y' then 'year'
-        when 'quarters', 'Q' then 'quarter'
-        when 'months', 'M' then 'month'
-        when 'weeks' , 'W'  then 'week'
-        when 'hours' , 'H' then 'hour'
-        when 'seconds', 'S' then 'second'
+      interval = switch @get('xAxisDomainType')
+        when 'Y' then 'year'
+        when 'Q' then 'quarter'
+        when 'M' then 'month'
+        when 'W'  then 'week'
+        when 'D' then 'day'
+        when 'H' then 'hour'
+        when 'S' then 'second'
       if interval is 'quarter'
         count = 3
         interval = 'month'
       (@_advanceMiddle(tick, interval, count) for tick in ticks)
-  .property 'xDomain', 'maxNumberOfLabels', 'centerAxisLabels', 'selectedInterval'
+  .property 'xDomain', 'maxNumberOfLabels', 'centerAxisLabels',
+    'xAxisDomainType'
```

Comment:
`xDomain` has `maxNumberOfLabels` as a property so you can delete `maxNumberOfLabels`